### PR TITLE
Ml2 59 loan display

### PR DESCRIPTION
**Pull request by mcstover**

### CODE REVIEW

File: src/pages/LandingPages/CorporateCampaign/CCLandingPage.vue [hunk 1]
```diff
@@ -78,14 +101,22 @@
...
+import _invokeMap from 'lodash/invokeMap';
+import _mapValues from 'lodash/mapValues';
+import _merge from 'lodash/merge';
+import basicLoanQuery from '@/graphql/query/basicLoanData.graphql';
 import { processPageContentFlat } from '@/util/contentfulUtils';
-import { validateQueryParams } from '@/util/campaignUtils';
+import { validateQueryParams, getPromoFromBasket } from '@/util/campaignUtils';
 import { lightHeader, lightFooter } from '@/util/siteThemes';
+import cookieStore from '@/util/cookieStore';
+// import paginationMixin from '@/plugins/pagination-mixin';
```
Comment (mcstover) [inline src/pages/LandingPages/CorporateCampaign/CCLandingPage.vue:112]: Started work on both a mixin and util file and will work on moving as much pagination related code as possibly into them a little later.

File: src/pages/LandingPages/CorporateCampaign/CCLandingPage.vue [hunk 2]
```diff
@@ -216,29 +296,171 @@ export default {
...
 				this.loadingPromotion = false;
 			}).catch(error => {
 				console.error(error);
 				this.loadingPromotion = false;
 			});
-		}
-	}
+		},
+		getPromoInformationFromBasket() {
+			// TEMPORARY query to gather credits and promo id from latest basket state
```
Comment (mcstover) [inline src/pages/LandingPages/CorporateCampaign/CCLandingPage.vue:349]: Upcoming GraphQL changes will allow us to remove this extra query.

File: src/pages/LandingPages/CorporateCampaign/CCLandingPage.vue [hunk 3]
```diff
@@ -216,29 +296,171 @@ export default {
...
+			// TEMPORARY Handling for patched in basket credits
+			// TODO Extract as utility to get promo id from basket credits
+			basketItems.then(({ data }) => {
+				const credits = data.shop?.basket?.credits?.values;
+				const targetPromo = credits.filter(credit => {
+					return credit.promoFund ? Object.keys(credit.promoFund).length > 0 : false;
+				});
+				// Primary PromoCampaign Query
+				// Future usage will not require the promoFundId relying only on the basket id
+				return getPromoFromBasket(targetPromo[0].promoFund?.id, this.apollo);
```
Comment (mcstover) [inline src/pages/LandingPages/CorporateCampaign/CCLandingPage.vue:385]: Initializes the new PromoCampign query using basket state

File: src/pages/LandingPages/CorporateCampaign/CCLandingPage.vue [hunk 4]
```diff
@@ -333,13 +555,29 @@ export default {
...
+}
+
+.cards-loan-card {
+	border-radius: 0.65rem;
+	box-shadow: 0 0.65rem $card-margin $card-half-space rgb(153, 153, 153, 0.1);
+	width: $card-width;
+	max-width: $max-card-width;
+	flex: 1 0 auto;
+	margin: $card-margin;
+}
```
Comment (mcstover) [inline src/pages/LandingPages/CorporateCampaign/CCLandingPage.vue:577]: Initially used this for layout purposes but do see some anomalies on the card and will have to fill in some more styles later.

File: src/pages/LandingPages/CorporateCampaign/CCLandingPage.vue [hunk 5]
```diff
@@ -216,29 +296,171 @@ export default {
...
+					return credit.promoFund ? Object.keys(credit.promoFund).length > 0 : false;
+				});
+				// Primary PromoCampaign Query
+				// Future usage will not require the promoFundId relying only on the basket id
+				return getPromoFromBasket(targetPromo[0].promoFund?.id, this.apollo);
+			}).then(response => {
+				// TODO Handle response and any potential errors
+				this.promoData = response.data?.shop?.promoCampaign;
+				// Initialize loan query + observer
+				this.activateLoanWatchQuery();
```
Comment (mcstover) [inline src/pages/LandingPages/CorporateCampaign/CCLandingPage.vue:390]: Setting up the watchQuery and watch will also fetch the initial loan query.

**Review by emuvente — APPROVED** [review]
:+1:

**Review by ryan-ludwig — APPROVED** [review]
👍

**Pull request merged**